### PR TITLE
Upgrade normalize.css to v8.0.1

### LINKED ISSUE #86629: rustdoc: upgrade or remove normalize.css
We're using a very old version of normalize.css (3.0.0, from 2014). In #82313 I tried to update to the latest, 8.0.1. But it was reverted in #82549 because of these regressions:

#82548
#82542

At the time we didn't worry about it much since we didn't need the update for anything. However, it turns out in #86589 that our old version of normalize.css is hiding the triangles on user-provided `<details>` tags, when it shouldn't be.

See these demos:

https://jsbin.com/jorowazuli/edit?html,output

https://jsbin.com/kecozayaka/edit?html,output

The right path forward is to figure out why upgrading normalize.css caused those regressions, fix the issues, and then do the upgrade. Alternately, we may conclude that we no longer need normalize.css, which would be an even nicer outcome.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #86629, this addresses #82548 and #82542 with tweaks. I expect that this changes the style *slightly* but shouldn't have any major changes.
Here's some changes I observed, I'd say they all are an improvement.

<details>
<summary>Before</summary>

before 1
before 2

</details>

<details>
<summary>After</summary>

after 1
after 2

</details>

r? @jsha

### CODE REVIEW

File: src/librustdoc/html/static/rustdoc.css [hunk 1]
```diff
@@ -200,10 +200,31 @@ summary {
...
+textarea {
+	color: inherit;
+	font: inherit;
+	margin: 0;
+}
+
+/* end tweaks for normalize.css 8 */
+
 details:not(.rustdoc-toggle) summary {
 	margin-bottom: .6em;
```
Comment (JohnTitor) [inline src/librustdoc/html/static/rustdoc.css:227]: NOTE: `margin-bottom` is still required.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Let's go with it and hope for the best I guess. Thanks!

@bors: r+

**Comment by JohnTitor** [comment]
At a glance, I've noticed two changes:

<details>
<summary>Beta</summary>

beta 1
beta 2

</details>

<details>
<summary>Nightly</summary>

nightly 1
nightly 2

</details>

The deprecation note width seems fine to me (actually, I messed up the screenshots on OP, sorry), but I'm happy to restore the original.

The problem is the second, I think it's a regression and we should fix. But somehow I couldn't find it on testing. I generated the docs via `./x.py doc library/std` but it's different from the actual?

@GuillaumeGomez I'd like to hear your thoughts.

**Comment by GuillaumeGomez** [comment]
Please reset the deprecation notice. As for the position of the labels, it comes from [this PR](https://github.com/rust-lang/rust/pull/86594), so nothing to change on your side.

@dns2utf8: Can we somehow fix the labels position while keeping them in the text by any chance?

> The problem is the second, I think it's a regression and we should fix. But somehow I couldn't find it on testing. I generated the docs via `./x.py doc library/std` but it's different from the actual?

Try emptying your cache? Otherwise the command line seems good to me.

**Comment by JohnTitor** [comment]
Alright, then I'll fix the first one later, thanks for checking!

**Comment by dns2utf8** [comment]
My changes were quite substantial and you must empty your css cache this time.
I am not 100% sure what you mean by "the second". Do you mean the shape of the coloured boxes for the "Deprecated" and similar labels?
They look like the ones currently on stable and they do not overlap with the current nightly
image

**Comment by GuillaumeGomez** [comment]
@dns2utf8 Yes, I was talking about the position. They are supposed to be positionned like the screenshot you provided. Do you think it's possible to keep them this way? If it requires too much changes, let's keep them as is.

**Comment by dns2utf8** [comment]
Moving the tags back to the right would be a big undertaking and to keep the benefits for mobile users and not reverting #86594
We could try to align the tags on the right side of the left column but I fear that could make it harder to read

**Comment by GuillaumeGomez** [comment]
We could apply a "flex" style so that the left column first item (so before the "Experimental" and other labels) has the same width for all items when not on mobile?

**Comment by dns2utf8** [comment]
A flex layout in the left column could work, yes. Let us open an issue to focus there :+1:

**Comment by GuillaumeGomez** [comment]
I let you do that then (odn't forget to ping me on it so I can add the tags). ;)

**Comment by JohnTitor** [comment]
After some investigating, turned out the first one also wasn't caused by this PR but #85970: https://github.com/rust-lang/rust/pull/85970/files#diff-7dc22a0530802d77c2f2ec9e834024a5657b6eab4055520fca46edc99a544413L855
If we re-add `display: table` to the `.stab` class, the width is restored. But I'm not sure if it was intentional or by accident.
@GuillaumeGomez Do you still want to restore it? If so, I'm happy to open a PR.

**Comment by dns2utf8** [comment]
I think the .stab problem has been solved already. However, I did not look into how exactly:
https://doc.rust-lang.org/nightly/std/primitive.array.html#method.zip
image

**Comment by GuillaumeGomez** [comment]
I just checked on stable to ensure we were doing the right thing and apparently we're not:

Screenshot from 2021-07-01 10-14-55

However, I kinda like it as it is...

**Comment by JohnTitor** [comment]
@GuillaumeGomez So, let's say we do not restore (shrink) it for now?

**Comment by GuillaumeGomez** [comment]
I'm good either way.

**Comment by JohnTitor** [comment]
Hmm alright, then I'll tweak it when someone issues it.

**Pull request merged**